package/tests/test_server.py

```diff
@@ -184,38 +184,64 @@ def test_no_browser(cli_runner):
 
 def test_load_file_outside_kedro_project(cli_runner, tmp_path):
     """
-    Check that running viz with `--load-file` flag works outside of a Kedro project
+    Check that running viz with `--load-file` flag works outside of a Kedro project.
     """
     filepath_json = str(tmp_path / "test.json")
     data = {"nodes": None, "edges": None, "tags": None}
     with open(filepath_json, "w") as f:
         json.dump(data, f)
 
-    result = cli_runner.invoke(
-        server.commands, ["viz", "--load-file", filepath_json]
-    )
+    result = cli_runner.invoke(server.commands, ["viz", "--load-file", filepath_json])
+    assert result.exit_code == 0, result.output
+
+
+@pytest.mark.usefixtures("patched_get_project_context")
+def test_save_file(cli_runner, tmp_path):
+    """
+    Check that running with `--save-file` flag saves pipeline JSON file in a specified path.
+    """
+    save_path = str(tmp_path / "test.json")
+
+    result = cli_runner.invoke(server.commands, ["viz", "--save-file", save_path])
     assert result.exit_code == 0, result.output
 
+    with open(save_path, "r") as f:
+        json_data = json.load(f)
+    assert json_data == EXPECTED_PIPELINE_DATA
+
+
+def test_no_top_level_key(cli_runner, tmp_path):
+    """
+    Check that top level keys are properly checked.
+    """
```

Comment:
I'd put all of these one-liner docstrings on one line, think that's more consistent with Google docstrings.
`"""blabla"""`